src/compiler/checker.ts

```diff
@@ -25724,24 +25724,33 @@ namespace ts {
                         some((<ClassDeclaration>node.parent).members, isInstancePropertyWithInitializer) ||
                         some(node.parameters, p => hasModifier(p, ModifierFlags.ParameterPropertyModifier));
 
-                    // Skip past any prologue directives to find the first statement
-                    // to ensure that it was a super call.
                     if (superCallShouldBeFirst) {
-                        const statements = node.body!.statements;
-                        let superCallStatement: ExpressionStatement | undefined;
+                        // Until we have better flow analysis, it is an error to place the super call within any kind of block or conditional
+                        // See GH #8277
+                        if (superCall.parent.parent !== node.body) {

```

Comment:
`(super())`?

Reply:
:+1: Added parenthesis diving with tests for `(super());` alone, before a `this.prop;`, and after a `this.prop;`.